routing/unified_edges.go

```diff
@@ -62,9 +65,18 @@ func (u *nodeEdgeUnifier) addPolicy(fromNode route.Vertex,
 		u.edgeUnifiers[fromNode] = unifier
 	}
 
+	// In case no payload size function was provided a graceful shutdown
+	// is requested, because this function is not used as intended.
+	if hopPayloadSizeFn == nil {
+		log.Criticalf("No payloadsize function was provided for the "+
+			"edge (chanid=%v) when adding it to the edge unifier "+
+			"of node: %v", edge.ChannelID, fromNode)
+	}
```

Comment:
return after the critical log otherwise we'll go on to line 79 and panic anyway? It takes some time for the signal to propagate through LND